### DISCUSSION TITLE
[SOLVED] PWA - Possibility of a cached response with getFullList when no internet connection.

### DISCUSSION THREAD

**Discussion by Mashiane** [pocketbase/pocketbase#4379] [Q&A]
Hi

I am working on a PWA deployed to pocketbase public folder. Everything is working fine when the internet is connected, all CRUD operations are perfect.

When I test my PWA offline, I am able to execute creation, updated and deletes, the thing is with reads, it seems the reading responses are retrieved from a cache as it never gets the database collection updated content when offline. The PWA works well in all aspects without the internet connection via the service worker, it's the reading from PocketBase of the latest collection information that does not seem to take place.

How can I ensure that I am getting the latest information from the database when working without an internet connection?

Thanks in advance.

**Comment by ganigeorgiev** [comment]
>  I test my PWA offline, I am able to execute creation, updated and deletes

I'm not sure that I understand what do you mean by this.

There is no support for offline mode at the moment (at least not out of the box) in the the official JS and Dart SDKs and internet connection is always required if you want to interact with the PocketBase server (_unless you are testing locally on the same device where the server process is running?_).

**Reply by ganigeorgiev** [reply]
> reading from PocketBase of the latest collection information that does not seem to take place.

Also note that there is no builtin server-side API at the moment that will retrieve only the last changes (new/updated/deleted records) that your local offline app is missing, if that is what you are trying to do.

**Reply by Mashiane** [reply]
Sorry I should have been clear that this is a local deployed environment. So yeah, I am testing on the same machine locally where the same server is.

For example, my collection is currently empty.

![image](https://github.com/pocketbase/pocketbase/assets/5799306/58fb884f-c72c-4035-a85e-6e627dfb8d90)

However, on my app each time I "refresh", ie. executing a getFullList its returning "old" records, which I am assuming are residing in some form of cache somewhere? I had noted this before with **fetch** and as a result I always append my querystring with "?" + DatetTime.Now for example in other projects when I use javascript fetch, for example fetching images. Im not sure what PocketBase uses internally

![image](https://github.com/pocketbase/pocketbase/assets/5799306/b17cdd51-c483-452d-acd9-951172e22b6d)

I would expect the same behavour to happen for example, for creating, updating and deleting, everything works as expected without an internet connection on my local deployment, however the updated, created records are not read from the db when executing the reads.

**Reply by ganigeorgiev** [reply]
I'm sorry, I still don't understand what do you mean.

> I always append my querystring with "?" + DatetTime.Now for example in other projects when I use javascript fetch, for example fetching images. Im not sure what PocketBase uses internally

We don't do any server or client side caching.
The JS SDK uses whatever global `fetch` is available (you even have an option to specify a custom one - [Custom request options](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#custom-request-options)).

If your app is returning cached responses then there is something in your app that does this but I don't know how to help you with that. If you are using Next.js note that they have cache enabled by default, for opting-out please refer to https://nextjs.org/docs/app/building-your-application/caching#opting-out-1.

**Reply by Mashiane** [reply]
Thank you so much. I also suspect that the issue is with the global **fetch**. Let me play around with the options and see what I can achieve.

My desktop PWA working offline on the local pocketbase deployment would be of great advantage. Thanks for pointing these options for me.

**Reply by Mashiane** [reply]
This has been resolved now. After adding the cache:     'no-store', on my SDK options, everything works as expected.
Perfect and thank you so much for your help.
